### PR TITLE
[efr32] call mbedtls_hardware_poll directly in otPlatEntropyGet()

### LINKED ISSUE #4649: [efr32] Entropy questions
Hi all. I have some questions regarding the efr32 platforms' entropy implementation.

1. Why is the ADC used instead of the TRNG in efr32mg12 and efr32mg13? I know that there is an erratum associated with the xg13 chip revisions A~C, but not from D onwards. The mg21 platform abstraction uses the TRNG. I'd like to propose to use the TRNG for all platforms, with the ADC only as fallback. This should free the ADC for the application to use.

2. The entropy collection code for the mg21 is as follows:
https://github.com/openthread/openthread/blob/fdcb8553b12216d53ed3c5249a97ebe7ff2e10b7/examples/platforms/efr32mg21/entropy.c#L40-L63
The `otPlatEntropyGet` function discards 3 out of 4 entropy bytes. Why? Couldn't we call directly `mbedtls_hardware_poll`?

3. In the mg21 code, shouldn't we check `ouputLen`? Specifically, seeing the mbedtls/sl_crypto/trng.c file, the `mbedtls_trng_poll` call could fail, setting `outputLen` to 0.

Thanks!

### PR DESCRIPTION
**Pull request by lmnotran**
Closes #4649

### CODE REVIEW

**Comment by jwhui** [comment]
/cc/ @dismirlian

**Review by romacdon — APPROVED** [review]

File: examples/platforms/efr32/src/entropy.c [hunk 1]
```diff
@@ -33,31 +33,29 @@
...
+        const uint16_t remaining = aOutputLength - outputLen;
+        partialLen               = 0;
+
+        // Non-zero return values for mbedtls_hardware_poll() signify an error has occurred
+        otEXPECT_ACTION(0 == mbedtls_hardware_poll(NULL, &aOutput[outputLen], remaining, &partialLen),
+                        error = OT_ERROR_FAILED);
+        otEXPECT_ACTION(partialLen > 0, error = OT_ERROR_FAILED);
     }
 
+    otEXPECT_ACTION(outputLen == aOutputLength, error = OT_ERROR_FAILED);
```
Comment (dismirlian) [inline examples/platforms/efr32/src/entropy.c:57]: Isn't this redundant? The above loop breaks when `outputLen == aOutputLength`. Also, it exits directly on error, right?

**Comment by dismirlian** [comment]
Thanks @lmnotran.

File: examples/platforms/efr32/src/entropy.c (hunk 1 above)
Comment (lmnotran) [reply examples/platforms/efr32/src/entropy.c:57]: You're definitely correct!

**Review by jwhui — APPROVED** [review]

**Pull request merged**